fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    async function getUser() {
+      try {
+        const resp = await fetch(
+          `https://jsonplaceholder.typicode.com/users/${userId}`
+        );
```

Comment:
You need to check resp.ok here. That is absolutely required for proper error handling with fetch.